### PR TITLE
Initial commit for Babelfish v3.4

### PR DESCRIPTION
**Pull request by basasairohan**
### Description

1. Update Github action scripts
2. Create 15_4 upgrade test folder.
3. Bump Babelfish version string in babelfish_version.h
4. Add upgrade script babelfishpg_tsql--3.3.0--3.4.0.sql

Signed-off-by: Sai Rohan Basa [bsrohan@amazon.com](mailto:bsrohan@amazon.com)

### Issues Resolved

[List any issues this PR will resolve]

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).please delete the guidelines after writing the PR description)]

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/template/version-branch-template.yml [hunk 1]
```diff
@@ -40,6 +40,9 @@
 '15.3':
   engine_branch: BABEL_3_2_STABLE__PG_15_3
   extension_branch: BABEL_3_2_STABLE
+'15.4':
```
Comment (rishabhtanwar29) [inline .github/template/version-branch-template.yml:46]: We need to add one new entry for current dev branches for previous release as well. Currently 14.9 above points to dev but we need to now change it to stable branches and then add new 14.10 version which should point to dev branches now.

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+LANGUAGE plpgsql;
+
+-- Please add your SQLs here
+
+-- Drops the temporary procedure used by the upgrade script.
+-- Please have this be one of the last statements executed in this upgrade script.
+DROP PROCEDURE sys.babelfish_drop_deprecated_object(varchar, varchar, varchar);
+
+-- Reset search_path to not affect any subsequent scripts
+SELECT set_config('search_path', trim(leading 'sys, ' from current_setting('search_path')), false);
```
Comment (rishabhtanwar29) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql]: Nit: add a new line.

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        GET STACKED DIAGNOSTICS error_msg = MESSAGE_TEXT;
+        raise warning '%', error_msg;
+    when dependent_objects_still_exist then --if 'drop view' statement fails
+        GET STACKED DIAGNOSTICS error_msg = MESSAGE_TEXT;
+        raise warning '%', error_msg;
+end
+$$
+LANGUAGE plpgsql;
+
+-- Please add your SQLs here
```
Comment (rishabhtanwar29) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql:34]: Let's include [this](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_3_X_DEV/contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql#L54) note here.

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.7.0--3.0.0.sql [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    select oid into table_oid from pg_class where relname = table_name collate sys.database_default and relnamespace = sys_schema;
+    select attcollation into att_coll from pg_attribute where attname = column_name collate sys.database_default and attrelid = table_oid;
+    if att_coll = default_coll_oid or att_coll = c_coll_oid then
+        update pg_attribute set attcollation = sys.babelfishpg_tsql_get_babel_server_collation_oid() where attname = column_name collate sys.database_default and attrelid = table_oid;
+    end if;
+END
+$$
+LANGUAGE plpgsql;
+
+-- please add your SQL here
```
Comment (rishabhtanwar29) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.7.0--3.0.0.sql]: Let's include [this](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_3_X_DEV/contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql#L54) note here.

File: .github/configuration/upgrade-test-configuration.yml [hunk 5]
```diff
@@ -103,4 +103,14 @@ upgrade-version: [[
     version: target.latest,
     upgrade-type: minor
   }
+],
+[
```
Comment (rishabhtanwar29) [inline .github/configuration/upgrade-test-configuration.yml]: This seems unnecessary, let's update the above one from 15.3 -> latest to 15.4 ->latest

File: test/JDBC/upgrade/15_4/schedule [hunk 6]
```diff
@@ -0,0 +1,450 @@
+# Schedule File for JDBC Test Framework for local run
```
Comment (rishabhtanwar29) [inline test/JDBC/upgrade/15_4/schedule:1]: We would need 14.10/schedule as well, which should be replica of latest/schedule from 14.latest

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql [hunk 7]
```diff
@@ -0,0 +1,41 @@
+-- complain if script is sourced in psql, rather than via ALTER EXTENSION
```
Comment (ahmed-shameem) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql:1]: Can you kindly make sure whether we need one upgrade script for babelfishpg_common extension as well or not?

File: .github/template/version-branch-template.yml (hunk 1 above)
Comment (basasairohan) [reply .github/template/version-branch-template.yml:46]: Added

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql (hunk 2 above)
Comment (basasairohan) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql]: Done

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql (hunk 3 above)
Comment (basasairohan) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql:34]: Done

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.7.0--3.0.0.sql (hunk 4 above)
Comment (basasairohan) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.7.0--3.0.0.sql]: Done

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql (hunk 7 above)
Comment (basasairohan) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql:1]: We can bump it up later when we need any SQL changes in the common extension

File: test/JDBC/upgrade/15_4/schedule (hunk 6 above)
Comment (basasairohan) [reply test/JDBC/upgrade/15_4/schedule:1]: Done

File: .github/configuration/upgrade-test-configuration.yml (hunk 5 above)
Comment (basasairohan) [reply .github/configuration/upgrade-test-configuration.yml]: Updated

**Review by rishabhtanwar29 — APPROVED** [review]

**Comment by rishabhtanwar29** [comment]
Please update the step 4 to reflect correct changes.

**Review by jsudrik — APPROVED** [review]

**Pull request merged**